redis/connection.py

```diff
@@ -1095,9 +1101,11 @@ def __init__(
         self.pid = os.getpid()
         self.path = path
         self.db = db
-        self.username = username
         self.client_name = client_name
-        self.password = password
+        self.credential_provider = credential_provider
+        if (username or password) and self.credential_provider is None:
+            # username and password backward compatibility
+            self.credential_provider = StaticCredentialProvider(username, password)
```

Comment:
Same comments about backward compat and mutual exclusivity here.

Reply:
added